### PR TITLE
CATALOG-9785: [update] Catalog, update all Brands V3 endpoints

### PR DESCRIPTION
**Pull request by NataliiaNefodova**
# [CATALOG-9785] Categories: REST Contract

## What changed?
* Updated v3 brands swagger scheme

## Release notes draft
* updated description for endpoints
* add responses code to endpoints
* deleted params "Content-Type" and "Accept" base https://swagger.io/docs/specification/describing-parameters/

ping @bigcommerce/team-catalog

[CATALOG-9785]: https://bigcommercecloud.atlassian.net/browse/CATALOG-9785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/catalog/brands_catalog.v3.yml [hunk 1]
```diff
@@ -274,8 +282,6 @@ paths:
         **Limits**
         - 30,000 brands per store limit
       operationId: createBrand
-      parameters:
-        - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:278]: Why was the Content-type parameter removed?

File: reference/catalog/brands_catalog.v3.yml [hunk 2]
```diff
@@ -523,8 +530,12 @@ paths:
         '204':
           description: ''
           content: {}
-    parameters:
-      - $ref: '#/components/parameters/Accept'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:527]: Why did you remove the Accept parameter?

File: reference/catalog/brands_catalog.v3.yml [hunk 3]
```diff
@@ -589,6 +606,10 @@ paths:
                   instance:
                     type: string
                 description: Error payload for the BigCommerce API.
+        '409':
+          $ref: '#/components/responses/Conflict'
+        '422':
+          $ref: '#/components/responses/ErrorWithDetailss'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:612]: ErrorWithDetails or ErrorWithDetailss

File: reference/catalog/brands_catalog.v3.yml [hunk 4]
```diff
@@ -598,8 +619,6 @@ paths:
 
         To update a brand image, send a request with an `image_url`.
       operationId: updateBrand
-      parameters:
-        - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:602]: Can you explain why you removed Content-type?

File: reference/catalog/brands_catalog.v3.yml [hunk 5]
```diff
@@ -971,8 +988,6 @@ paths:
 
         **Note:** The maximum number of metafields allowed on each order, product, category, variant, or brand is 250 per client ID. For more information, see [Platform Limits](https://support.bigcommerce.com/s/article/Platform-Limits) in the Help Center.
       operationId: createBrandMetafield
-      parameters:
-        - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:975]: same question

File: reference/catalog/brands_catalog.v3.yml [hunk 6]
```diff
@@ -1092,7 +1107,6 @@ paths:
                     type: string
       x-codegen-request-body-name: Metafield
     parameters:
-      - $ref: '#/components/parameters/Accept'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1095]: same question

File: reference/catalog/brands_catalog.v3.yml [hunk 7]
```diff
@@ -1157,8 +1171,6 @@ paths:
       summary: Update a Brand Metafield
       description: "Updates a brand metafield.\n\n**Required Fields**  \n* none\n\n**Read-Only Fields**\n* id\n* These fields can only be modified by the app (API credentials) that created the metafield:\n\t* namespace\n\t* key\n\t* permission_set\n\n**Usage Notes**\n* Attempting to modify `namespace`, `key`, and `permission_set` fields using a client ID different from the one used to create those metafields will result in a 403 error message.\n* The maximum number of metafields allowed on each order, product, category, variant, or brand is 250 per client ID. For more information, see [Platform Limits](https://support.bigcommerce.com/s/article/Platform-Limits) in the Help Center."
       operationId: updateBrandMetafield
-      parameters:
-        - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1161]: same question

File: reference/catalog/brands_catalog.v3.yml [hunk 8]
```diff
@@ -1224,7 +1236,6 @@ paths:
           description: ''
           content: {}
     parameters:
-      - $ref: '#/components/parameters/Accept'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1227]: same question

File: reference/catalog/brands_catalog.v3.yml [hunk 9]
```diff
@@ -1243,8 +1254,6 @@ paths:
 
         Only one image at a time can be created. To update a brand image, use the [Update a brand](/docs/rest-catalog/brands#update-a-brand) endpoint and an `image_url`.
       operationId: createBrandImage
-      parameters:
-        - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1247]: same question

File: reference/catalog/brands_catalog.v3.yml [hunk 10]
```diff
@@ -1337,7 +1346,6 @@ paths:
           description: ''
           content: {}
     parameters:
-      - $ref: '#/components/parameters/Accept'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1340]: same

File: reference/catalog/brands_catalog.v3.yml [hunk 11]
```diff
@@ -1371,8 +1379,6 @@ paths:
         - Batch metafields
       description: Create multiple metafields.
       operationId: createBrandsMetafields
-      parameters:
-      - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1375]: same

File: reference/catalog/brands_catalog.v3.yml [hunk 12]
```diff
@@ -1414,8 +1420,6 @@ paths:
         - Batch metafields
       description: Create multiple metafields.
       operationId: updateBrandsMetafields
-      parameters:
-      - $ref: '#/components/parameters/ContentType'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1418]: same

File: reference/catalog/brands_catalog.v3.yml [hunk 13]
```diff
@@ -1480,8 +1484,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_DELETE'
-    parameters:
-      - $ref: '#/components/parameters/Accept'
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:1484]: same

File: reference/catalog/brands_catalog.v3.yml [hunk 14]
```diff
@@ -2437,13 +2453,58 @@ components:
...
-    General207Status:
-      description: Multi-status. Multiple operations have taken place and the status for each operation can be viewed in the body of the response. Typically indicates that a partial failure has occurred, such as when a `POST` or `PUT` request is successful, but saving the URL or inventory data has failed.
-
+    Error:
+      description: ''
       content:
         application/json:
           schema:
             $ref: '#/components/schemas/error_Base'
+    ErrorWithDetailss:
```
Comment (bc-traciporter) [inline reference/catalog/brands_catalog.v3.yml:2462]:
```suggestion
    ErrorWithDetails:
```

**Review by bc-traciporter — COMMENTED** [review]
I left comments.

File: reference/catalog/brands_catalog.v3.yml (hunk 1 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 2 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:527]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 3 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Renamed to ErrorWithDetails

File: reference/catalog/brands_catalog.v3.yml (hunk 5 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 6 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:1095]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 7 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 8 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:1227]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 9 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 10 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:1340]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 11 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:1375]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 12 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml:1418]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 13 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Returned to code

File: reference/catalog/brands_catalog.v3.yml (hunk 14 above)
Comment (NataliiaNefodova) [reply reference/catalog/brands_catalog.v3.yml]: Renamed

**Comment by bc-traciporter** [comment]
Hi @NataliiaNefodova your changes look great.  I would like to merge PR 257 first and then compare it with your changes.

**Comment by bc-traciporter** [comment]
@NataliiaNefodova 
Can you add the `Accept` parameter to the Create a brand endpoint? It is missing.
- $ref: '#/components/parameters/Accept'

**Review by bc-traciporter — DISMISSED** [review]
Thanks for adding the Accept parameter.
Thank you for submitting this PR.

File: reference/catalog/brands_catalog.v3.yml [hunk 15]
```diff
@@ -335,7 +340,7 @@ paths:
                 custom_url:
                   title: Custom Url Brand
                   type: object
-                  description: The custom URL for the brand on the storefront.
+                  description: The custom URL for the brand on the storefront. If not provided - the URL will be autogenerated from the brand name.
```
Comment (bc-andreadao) [inline reference/catalog/brands_catalog.v3.yml:343]:
```suggestion
                  description: The custom URL for the brand on the storefront. If not provided, the URL will be autogenerated from the brand name.
```

**Review by bc-andreadao — COMMENTED** [review]
I left some comments, looks good

**Review by bc-andreadao — APPROVED** [review]
